### DISCUSSION TITLE
How can I submit mail to port 25 for DMS to send?

### DISCUSSION THREAD

**Discussion by borisservo** [docker-mailserver/docker-mailserver#3995] [Q&A] [answered]
when using port 25 I'm receiving the following error.

```
May  1 21:47:24 mail postfix/postscreen[4599]: CONNECT from [172.20.0.3]:35012 to [172.20.0.2]:25
May  1 21:47:24 mail postfix/postscreen[4599]: PASS OLD [172.20.0.3]:35012
May  1 21:47:24 mail postfix/smtpd[4600]: connect from passbolt_app.dockermail_default[172.20.0.3]
May  1 21:47:24 mail postfix/smtpd[4600]: disconnect from passbolt_app.dockermail_default[172.20.0.3] ehlo=1 auth=0/1 quit=1 commands=2/3
```

I'm using port 25 to send messages from apps since I'm struggling to get them to work with port 465 and I need a temporary solution.

**Comment by glime-ai** [comment]
**Please note** that this response was created using generative AI, leveraging previously available information from the repository. This information may be helpful. However, it is our strong recommendation that you carefully consider the information before blindly applying it.

---

To assist you effectively, could you please provide the specific error message you are encountering when using port 25? This will help in diagnosing the issue more accurately.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=0d5b86fb7eea4a54acac405212c29aba&utm_medium=github).

**Comment by polarathene** [comment]
Port 25 is not authorized for mail submission, only delivery.

You have ports 465 or 587 to authenticate against when you want to provide mail for DMS to send.

- Create a DMS account within the container with the command `setup email add user@example.com secret-password`
- On your mail client, connect to DMS FQDN (eg: `mail.example.com`), you can use whatever sender address you like, but the credentials to connect should be the DMS account `user@example.com` with password `secret-password`, or whatever you chose instead.
- Make sure you connect to port 587 with STARTTLS encryption selected, or regular TLS encryption for port 465. The password should be sent as plain text, some mail clients call this "None".
- Provided your TLS certificate for the DMS FQDN you're connecting to is valid and correctly configured, the mail client should connect to DMS and authenticate successfully. Your mail will then be accepted and sent.

If you search our issues or discussions, you will find advice on how to enable port 25 for sending mail without authentication, but it would be better for you to troubleshoot the above guidance so that you can send mail properly.

**Reply by borisservo** [reply]
Hello polarathene,

Email is already setup and working I have it in Thunderbird working.
I want to use the DockerMail to send emails from others apps per example recover password, sign in confirmation, etc.

All the apps are in container inside the same docker, when trying to send I constantly get this error.

DockerMail -> port 465 TLS
```
May  4 21:46:01 mail postfix/submission/smtpd[712300]: warning: hostname dockerfrontend does not resolve to address 172.20.0.1
May  4 21:46:01 mail postfix/submission/smtpd[712300]: connect from unknown[172.20.0.1]
May  4 21:46:01 mail postfix/submission/smtpd[712300]: SSL_accept error from unknown[172.20.0.1]: -1
May  4 21:46:01 mail postfix/submission/smtpd[712300]: warning: TLS library problem: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca:../ssl/record/rec_layer_s3.c:1562:SSL alert number 48:
May  4 21:46:01 mail postfix/submission/smtpd[712300]: lost connection after STARTTLS from unknown[172.20.0.1]
May  4 21:46:01 mail postfix/submission/smtpd[712300]: disconnect from unknown[172.20.0.1] ehlo=1 starttls=0/1 commands=1/2
May  4 21:47:02 mail postfix/submission/smtpd[712300]: warning: hostname dockerfrontend does not resolve to address 172.20.0.1
May  4 21:47:02 mail postfix/submission/smtpd[712300]: connect from unknown[172.20.0.1]
May  4 21:47:02 mail postfix/submission/smtpd[712300]: SSL_accept error from unknown[172.20.0.1]: -1
May  4 21:47:02 mail postfix/submission/smtpd[712300]: warning: TLS library problem: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca:../ssl/record/rec_layer_s3.c:1562:SSL alert number 48:
May  4 21:47:02 mail postfix/submission/smtpd[712300]: lost connection after STARTTLS from unknown[172.20.0.1]
May  4 21:47:02 mail postfix/submission/smtpd[712300]: disconnect from unknown[172.20.0.1] ehlo=1 starttls=0/1 commands=1/2
```

Passbolt -> port 465 TLS
```
2024-05-04 21:47:02 warning: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:0A000086:SSL routines::certificate verify failed
warning: 2 :: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:0A000086:SSL routines::certificate verify failed on line 490 of /usr/share/php/passbolt/vendor/cakephp/cakephp/src/Network/Socket.php
2024/05/04 21:47:02 [info] 100#100: *8337 client closed connection while waiting for request, client: 172.16.0.254, server: 0.0.0.0:443
172.16.0.254 - - [04/May/2024:21:47:02 +0000] "GET /auth/is-authenticated.json?api-version=v2 HTTP/1.1" 200 277 "https://passbolt.example.com:9443/app/administration/smtp-settings" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
```

DockerMail -> port 587 TLS
```
May  4 22:00:01 mail postfix/submission/smtpd[714623]: warning: hostname dockerfrontend does not resolve to address 172.20.0.1
May  4 22:00:01 mail postfix/submission/smtpd[714623]: connect from unknown[172.20.0.1]
May  4 22:00:01 mail postfix/submission/smtpd[714623]: SSL_accept error from unknown[172.20.0.1]: -1
May  4 22:00:01 mail postfix/submission/smtpd[714623]: warning: TLS library problem: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca:../ssl/record/rec_layer_s3.c:1562:SSL alert number 48:
May  4 22:00:01 mail postfix/submission/smtpd[714623]: lost connection after STARTTLS from unknown[172.20.0.1]
May  4 22:00:01 mail postfix/submission/smtpd[714623]: disconnect from unknown[172.20.0.1] ehlo=1 starttls=0/1 commands=1/2
```

Passbolt  -> port 587 TLS
```
2024-05-04 22:00:01 warning: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:0A000086:SSL routines::certificate verify failed
warning: 2 :: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:0A000086:SSL routines::certificate verify failed on line 490 of /usr/share/php/passbolt/vendor/cakephp/cakephp/src/Network/Socket.php
2024-05-04 22:00:02,350 INFO reaped unknown pid 3560 (exit status 0)
2024-05-04 22:00:02,350 INFO reaped unknown pid 3562 (exit status 0)
2024-05-04 22:00:02,350 INFO reaped unknown pid 3564 (exit status 0)
2024-05-04 22:00:02,350 INFO reaped unknown pid 3566 (exit status 0)
2024-05-04 22:00:02,350 INFO reaped unknown pid 3568 (exit status 0)
2024-05-04 22:00:02,351 INFO reaped unknown pid 3570 (exit status 0)
2024-05-04 22:00:02,351 INFO reaped unknown pid 3572 (exit status 0)
2024-05-04 22:00:02,351 INFO reaped unknown pid 3575 (exit status 0)
```
I forgot to put the compose file to check if I'm doing something wrong. One thing to keep in mind is the way I have the network designed.
I have pfSense with haProxy for webservices, but DockerMail is not passing by the proxy is going directly to container. I had to input the certificates directly in the compose file.

```yaml
services:
  mailserver:
    image: ghcr.io/docker-mailserver/docker-mailserver:latest
    container_name: mailserver
    # Provide the FQDN of your mail server here (Your DNS MX record should point to this value)
    hostname: mail.example.com
    # Automatic cleanup (removes older image pulls from wasting disk space):
    environment:
      - SSL_TYPE=manual
      - ENABLE_RSPAMD=1
      - ENABLE_CLAMAV=1
      - ENABLE_FAIL2BAN=1
      - ONE_DIR=1
      - POSTFIX_MESSAGE_SIZE_LIMIT=26214400
      - ENABLE_OPENDKIM=0
      - ENABLE_OPENDMARC=0
      - ENABLE_POLICYD_SPF=0
      - ENABLE_AMAVIS=0
      - ENABLE_SPAMASSASSIN=0
      - SSL_CERT_PATH=/tmp/dms/custom-certs/example.com.crt
      - SSL_KEY_PATH=/tmp/dms/custom-certs/example.com.key
    env_file: mailserver.env
 
    # More information about the mail-server ports:
    # https://docker-mailserver.github.io/docker-mailserver/latest/config/security/understanding-the-ports/
    # To avoid conflicts with yaml base-60 float, DO NOT remove the quotation marks.
    ports:
        #- "25:25"    # SMTP  (explicit TLS => STARTTLS, Authentication is DISABLED => use port 465/587 instead)
        #- "143:143"  # IMAP4 (explicit TLS => STARTTLS)
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
      - "993:993"  # IMAP4 (implicit TLS)
    volumes:
     - /docker/certificates/:/tmp/dms/custom-certs/:ro
     - ./docker-data/dms/mail-data/:/var/mail/
     - ./docker-data/dms/mail-state/:/var/mail-state/
     - ./docker-data/dms/mail-logs/:/var/log/mail/
     - ./docker-data/dms/config/:/tmp/docker-mailserver/
     - /etc/localtime:/etc/localtime:ro

    restart: unless-stopped
    stop_grace_period: 1m
    cap_add:
     - NET_ADMIN
    healthcheck:
      test: "ss --listening --tcp | grep -P 'LISTEN.+:smtp' || exit 1"
      timeout: 3s
      retries: 0
```
Thanks in advance for the assist.

**Reply by polarathene** [reply]
> `warning: hostname dockerfrontend does not resolve to address 172.20.0.1`

This is likely because the connection is going through the docker network gateway (`172.x.x.1`), so it appears to be coming through a different IP. Since you have a reverse-proxy in use, this will be an issue regardless with retaining the IP. If you can use Traefik or similar with PROXY protocol support, you can preserve the original client IP if you'd like.

---

> `error:0A000086:SSL routines::certificate verify failed on line 490 of /usr/share/php/passbolt/vendor/cakephp/cakephp/src/Network/Socket.php`
> `warning: TLS library problem: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca:../ssl/record/rec_layer_s3.c:1562:SSL alert number 48`

These two similar errors are about the same issue. They cannot verify your custom cert as the CA is not known. The client (containers?) would need to have the CA used to sign your certificate in their trust stores to trust the certificate you've configured for DMS. Normally this is a non-issue as your certificate is signed by a CA already in the trust store like LetsEncrypt.

If you are new to this, you'll want to gain a better understanding of how to work with your own internal certificates. This isn't likely to be a DMS specific issue. Ensure that you're not using self-signed certificates (_no root CA used to establish trust_), once you've provisioned the certs correctly and any client has the signing CA in their trust stores they should be able to verify without an issue.

---

### Some guidance for troubleshooting along the way

In the DMS test suite, [we use `openssl` command to verify TLS is working correctly](https://github.com/docker-mailserver/docker-mailserver/blob/e2c2a22dcf2d8adc12f303b84b694945e8dc072e/test/tests/parallel/set1/tls/manual.bats#L92-L107), but point to the custom root CA certificate so that it can be verified. You should be able to run something like this:

```console
timeout 1 openssl s_client -connect mail.example.com:587 -starttls smtp -CAfile '/path/to/ca-cert.pem'
```

You can run that locally or in the container (_with `docker exec -it mailserver paste-command-here`_), the `CAfile` path needs to be valid for where the command is running. So if it's on the Docker host, the path is outside of the container. If you run it from within any container, that path should be referring to one in your container, such as a volume mount.

That should then have some output that includes `Verification: OK` if successful or `Verification error` if still failing. If it's successful, removing the `-CAfile` should then fail it, which is the scenario you're experiencing.

After that, you'll want to learn how to add your CA cert to your trust stores. This varies by OS distro, so might differ per container you have and you may need to run some commands to update the trust store beyond adding your CA cert there.

When you've done that successfully, that `openssl` command above without `-CAfile` will now be successful instead of failing, it should have no trouble finding the correct CA cert to verify your leaf cert for DMS. When that is successful, the errors you shared from other clients should also disappear 👍

**Reply by borisservo** [reply]
Hello @polarathene,

Thank you for the information, there are a couple of things that are not clear to me.

I'm using several containers inside the same docker, if I insert the `/etc/hosts` the local IP and the hostnames of the containers it shouldn't go to the proxy, it shouldn't even exit the host and beyond that it shouldn't even leave the vLAN since the DNS server has the hostnames of the containers and they are pointing to the docker server.
I might be wrong, but I do believe that proxy doesn't make a difference, since it should never hit the proxy.

![Screenshot from 2024-05-05 10-53-57](https://github.com/docker-mailserver/docker-mailserver/assets/15837056/5c7beb71-bf14-4686-b4ca-08cbc18ca416)

```
$ docker network ls
NETWORK ID     NAME                 DRIVER    SCOPE
9a496873b999   bridge               bridge    local
959349fdd863   docker_gwbridge      bridge    local
7525e9966b97   dockermail_default   bridge    local
817b6dc5dbf9   host                 host      local
yecracgobuil   ingress              overlay   swarm
93f01c6b3291   joplin_default       bridge    local
iu1790rijpcu   maria-db             overlay   swarm
06edf3e27879   none                 null      local
9dcfc62efb12   passbolt_default     bridge    local
b466nf6dddye   postgres-db          overlay   swarm
```

Is there a way in DockerMail to ignore TLS if connections are coming from private IPs or at least a whitelist to add the IP.
As you said an it makes sense, it is coming from an internal IP and those internal IPs are from the docker container network.

Another thing that I want to check, is if all the apps and DockerMail are in the same docker network does that bypass the TLS requirements?

I understand how certificates work, but this is my first time working with LetsEncrypt certificates not being run by the OS and yes by the proxy. Running by the OS it will insert the CA in the `/usr/local/share/ca-certificates/`.
I scripted to copy the certificates over to the docker every time they are renewed.

From my understanding the apps, mail servers, API, etc, should have the full chain certificate, usually a .pem, but in the variables for the several docker apps it seems that every app is only making "gets" to the .crt and .key, but in theory the .crt is also the CA which was copied to the trusted store.

```
sudo cp local-ca.crt /usr/local/share/ca-certificates
sudo update-ca-certificates

```
https://ubuntu.com/server/docs/install-a-root-ca-certificate-in-the-trust-store
https://ubuntu.com/server/docs/certificates
```
/docker/certificates$ ls -la
total 40
drwxrwxrwx   2 user docker 4096 Apr 28 16:40 .
drwxrwxr-x+ 10 user   docker 4096 Apr 30 23:19 ..
-rwxrwxrwx   1 user docker 5254 Apr 27 21:53 example.com.all.pem
-rwxrwxrwx   1 user docker 1826 Apr 27 21:53 example.com.ca
-rwxrwxrwx   1 user docker 1753 Apr 27 21:53 example.com.crt
-rwxrwxrwx   1 user docker 3579 Apr 27 21:53 example.com.fullchain
-rwxrwxrwx   1 user docker 1675 Apr 27 21:53 example.com.key
```
And yes the CA in theory was uploaded correctly to the trust store.

```
/docker/certificates$ sudo cp example.com.crt /usr/local/share/ca-certificates
/docker/certificates$ sudo update-ca-certificates
Updating certificates in /etc/ssl/certs...
1 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
```

For this reason I need a bit of help understanding how to make this viable without the certificates if private networking is in use or if all containers are in the same docker network.

Only outgoing emails, communications, webservices or API are mandatory  to have certificate signing, that the proxy handles it.
A thing that also might be creating some constrains is the use of wildcard certificate for all `domain.tld` instead of `subdomain.domain.tld`.

In hopes this helps to have a cleared picture if what I'm facing and what I wish to accomplish.

**Reply by polarathene** [reply, marked as answer]
Sorry about the 2 month delay in a response, I missed this 😓 (_I realize this is probably too late and you've probably figured it out or switched to another alternative_)

I haven't had time to go through the prior discussion again, so I might be repeating myself here or missing some context, sorry about that.

---

> I'm using several containers inside the same docker, if I insert the `/etc/hosts` the local IP and the hostnames of the containers it shouldn't go to the proxy, it shouldn't even exit the host

You shouldn't manually adjust `/etc/hosts` in a container AFAIK as Docker itself manages that file, you might run into issues? Docker does provide a CLI option or Compose setting to add additional hosts to `/etc/hosts`:

```console
$ docker run --rm -it --hostname example.test --add-host mail.example.test:1.2.3.4 alpine cat /etc/hosts

127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
1.2.3.4 mail.example.test
172.17.0.5      example.test example
```

> beyond that it shouldn't even leave the vLAN since the DNS server has the hostnames of the containers and they are pointing to the docker server.

Docker has it's own internal embedded DNS that it'll use to route to containers within the network (_via compose service name, hostname, container name, etc_). **EDIT:** I might be mistaken, and those are all added to `/etc/hosts` as shown above.

The daemon can have a default DNS service configured, I can't recall if that is the equivalent to explicitly setting one on a container (`--dns`), but the container by default (_without `--dns` set_) will point to the embedded Docker DNS server to query first, and fallback to another DNS server for external queries. That does have some limitations in my experience where some DNS operations are faulty unless using an explicit DNS service to bypass the embedded one:

```console
$ docker run --rm -it --dns 1.1.1.1 alpine cat /etc/resolv.conf

# Generated by Docker Engine.
# This file can be edited; Docker Engine will not make further changes once it
# has been modified.

nameserver 1.1.1.1

# Based on host file: '/etc/resolv.conf' (legacy)
# Overrides: [nameservers]
```

So maybe keep that in mind as a potential gotcha to be aware of

---

> Is there a way in DockerMail to ignore TLS if connections are coming from private IPs or at least a whitelist to add the IP.
> As you said an it makes sense, it is coming from an internal IP and those internal IPs are from the docker container network.

Off the top of my head, I think you can do something like this but it might vary by service (Dovecot, Postfix, Rspamd, etc).

We do have a `PERMIT_DOCKER` ENV you can try, but the current implementation has some issues (_there is a related TODO issue open for more info_). That feature will relax security on network scope such as by an entire subnet or connected interfaces/networks, I don't encourage it much due to the current implementation caveats, and you'd really want to ensure nothing external gets masqueraded as the Docker network gateway IP (eg: IPv6 connections routed to IPv4 in container).

It will allow you to bypass more than just TLS check however.

> Another thing that I want to check, is if all the apps and DockerMail are in the same docker network does that bypass the TLS requirements?

I think I just covered that. TLS shouldn't be necessary for port 25 to receive or send.

It is expected for the authenticated ports to ensure your credentials aren't sent unencrypted over the network. If that isn't a concern for you, you could choose not to use TLS at all.

--

> I understand how certificates work, but this is my first time working with LetsEncrypt certificates not being run by the OS and yes by the proxy. Running by the OS it will insert the CA in the `/usr/local/share/ca-certificates/`. I scripted to copy the certificates over to the docker every time they are renewed.
> 
> From my understanding the apps, mail servers, API, etc, should have the full chain certificate, usually a .pem, but in the variables for the several docker apps it seems that every app is only making "gets" to the .crt and .key, but in theory the .crt is also the CA which was copied to the trusted store.

We have fairly good coverage on this topic with our docs? Along with guides for automating renewals. Is there something missing?

For LetsEncrypt, be sure to handle this correctly. We have docs on this too regarding mounting a volume from the host so that it can resolve symlinks correctly (_common config mistake from users_).

`/etc/ssl/certs` (_or similar, varies by distro_) should already have the LetsEncrypt root CA certs, I assume `/usr/local/share/ca-certificates` is for your own custom root CA certs, not LE specific?

It's been a while since I refreshed on this topic, but DMS should only need the custom root CA certs added if it has any internal service making a request where that trust needs to be verified, which I think would be to a service external of DMS (_eg: another DMS instance acting as a relay host_). Otherwise, you just need to provide the public cert provisioned for DMS to use by that CA.

Any client connecting to DMS would then need that custom CA root cert in the clients trust store (_either the client software supports it's own trust store, or it uses the one from the OS_), and when connecting to DMS, the public cert will be provided (_this should contain any additional intermediate certs needed for the client to resolve the trust chain_).

You can find my docs on how we [provisioned certs for DMS test suite](https://github.com/docker-mailserver/docker-mailserver/tree/9175424d0f3c67622c030311cc720d7477f1c2b6/test/files/ssl/example.test). The Smallstep CLI tool makes it quite easy to provision the root CA and leaf cert files. You could add additional intermediate certs into the mix there too if you need to test configuring for such.

We use those cert files for test coverage in:
- [`SSL_TYPE=letsencrypt`](https://github.com/docker-mailserver/docker-mailserver/blob/9175424d0f3c67622c030311cc720d7477f1c2b6/test/tests/parallel/set1/tls/letsencrypt.bats)
- [`SSL_TYPE=manual`](https://github.com/docker-mailserver/docker-mailserver/blob/9175424d0f3c67622c030311cc720d7477f1c2b6/test/tests/parallel/set1/tls/manual.bats)
- [TLS helper](https://github.com/docker-mailserver/docker-mailserver/blob/9175424d0f3c67622c030311cc720d7477f1c2b6/test/helper/tls.bash) script used by those tests.

While the container startup configures TLS certs for Dovecot and Postfix with [this helper script](https://github.com/docker-mailserver/docker-mailserver/blob/9175424d0f3c67622c030311cc720d7477f1c2b6/target/scripts/helpers/ssl.sh#L31-L57). The public key (cert) assigned to these services to use, can contain the full chain, but for it to be possible to verify the client itself still needs the CA cert in it's trust store, it can't solely trust DMS as the only source of information to verify properly, so this certificate only needs to provide the details for DMS cert that the client wouldn't have (the leaf cert).

---

> For this reason I need a bit of help understanding how to make this viable without the certificates if private networking is in use or if all containers are in the same docker network.

Is there a reason you can't provision certificates for these connections to use instead?

> Only outgoing emails, communications, webservices or API are mandatory to have certificate signing, that the proxy handles it.

How would you define outgoing mails? A mail client submits mail to DMS to send outbound, are you expecting the proxy to handle TLS correctly for this mail client connecting to DMS, with that mail client not being in the private network or as a container on that docker network?

Do you already have the proxy handling this for you?
- AFAIK you can't have STARTTLS used (eg ports 25, 587) if you're terminating TLS at the proxy, simply because it was already a TLS connection from the client, if the proxy even supports STARTTLS in this manner I'm not sure what value it would add.
- Likewise, having a STARTTLS connection from the client or even TLS, and then having the proxy connect to DMS over TLS would not work either, since both of these would have DMS configured with certificates which kinda defeats the purpose of terminating TLS at the proxy.

Your proxy should just pass the TLS connection over to DMS to terminate. You can still provision the certificate via the proxy if that works for you, and have DMS use that. Alternative just bypass the proxy and expose the DMS ports directly, if you don't have more than one DMS instance, you probably don't have much use for the proxy. If you insist on the proxy, keep in mind of the client IP caveats this causes, you'll want to use the PROXY protocol.

---

> In hopes this helps to have a cleared picture if what I'm facing and what I wish to accomplish.

I was a little confused, but that's likely my fault for not going through the full discussion again.

Unfortunately I lack the time for assisting you better, but I do know that we have a working proxy guide with Traefik and PROXY protocol.
- This allowed for private / internal docker network to function with separate ports for Postfix and Dovecot, while external connections went through the proxy (_relying on PROXY protocol_).
- With that approach, you don't need to expose/publish the real mail ports (25, 465, 587) from the DMS container, and you can probably use something like `PERMIT_DOCKER` to trust the private network this way (_test that it functions as expected for untrusted and trusted clients_).

Regarding the TLS requirement, if verification is the only concern you want to ignore, I think your mail client should allow you to continue with TLS connections that aren't trusted. That's a bit better than trying to bypass TLS to rely on unencrypted connections for trusted clients?

**Reply by borisservo** [reply]
hello @polarathene,

Thank you for the explanation. I've decided to simplify the setup.

Instead of using the proxy, I've configured NAT rules and directly implemented Let's Encrypt in the docker-compose.

I initially wanted to experiment with new configurations for a more refined setup, but it proved to be too challenging and after extensive research, I couldn't find a solution.

To enhance security, I placed the containers that need to send emails on the same Docker network as docker-mail and opened port 25. I created a network specifically for this purpose. While port 25 is generally considered a security risk, using it internally should pose no issues.

Your explanation was exactly what I needed. If this were for a professional or enterprise mail server, I would need to reconfigure based on the new information. However, since this is for personal use, everything is working fine without being blocked by providers.

Thank you again for your time. This will likely be useful for someone facing the same scenario I did.
